### PR TITLE
New component: Toolbar (#412)

### PR DESCRIPTION
**Pull request by jjenzz**

### CODE REVIEW

**Comment by jjenzz** [comment]
@Nazeh Just a heads up that we had tried to get this into our planned release today but while testing it again locally, Benoit noticed that the roving index prevents arrow keys from working inside the input. 

This is a bit of an accessibility concern which led us to test if our other components would work (e.g. `RadioGroup`) and we spotted some roving index issues. With the potential of these conflicts in mind, we decided to go with the following changes:

- `ToolbarItem` renamed to `ToolbarButton`. Any of our components that provide a `Trigger` can do `<ToolbarButton as={Component.Trigger}>` as you had already demonstrated in the stories.
- Added `ToolbarRadioGroup` and `ToolbarRadioItem` components. These allow `Toolbar` to control roving focus.
- Added a `ToolbarLink` part so that consumers get the space key functionality you added to the story out of the box.
- Removed the `input` example from the story. After researching, this feels like a bit of an anti-pattern but we're not entirely sure. For now, we're not going to promote the idea anywhere until we can explore it further. We may provide a `ToolbarInput` component later with more keyboard accessibility bindings.
- Removed `ToolbarGroup`. It didn't provide any accessibility/functional enhancements and seems to be covered by `ToolbarRadioGroup` and `ToggleButtonGroup` that we have in PR atm. 

If you have an questions/change requests/further ideas, please do let us know 🙂

File: packages/react/radio-group/src/RadioGroup.tsx [hunk 1]
```diff
@@ -73,24 +76,27 @@ const RadioGroup = React.forwardRef((props, forwardedRef) => {
...
-      <RovingFocusGroup loop>
-        <Primitive
-          {...groupProps}
-          selector={selector}
-          ref={forwardedRef}
-          role="radiogroup"
-          aria-labelledby={labelledBy}
-        />
-      </RovingFocusGroup>
+      {rovingFocus ? <RovingFocusGroup loop>{primitive}</RovingFocusGroup> : primitive}
```
Comment (jjenzz) [inline packages/react/radio-group/src/RadioGroup.tsx:99]: I was torn between doing this or adding a `ConditionalWrap` component. Which do we prefer?

Comment (benoitgrelard) [reply packages/react/radio-group/src/RadioGroup.tsx:99]: Historically we've stuck with this for now. I'm happy to keep this.

File: packages/react/radio-group/src/RadioGroup.tsx [hunk 2]
```diff
@@ -103,58 +109,89 @@ RadioGroup.displayName = RADIO_GROUP_NAME;
...
+  );
+}) as RadioGroupItemPrimitive;
+
+type RadioGroupRovingFocusItemOwnProps = Polymorphic.OwnProps<typeof RadioGroupItemImpl>;
+type RadioGroupRovingFocusItemPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof RadioGroupItemImpl>,
+  RadioGroupRovingFocusItemOwnProps
+>;
+
+const RadioGroupRovingFocusItem = React.forwardRef((props, forwardedRef) => {
```
Comment (benoitgrelard) [inline packages/react/radio-group/src/RadioGroup.tsx:133]: I like this new separation now, and seeing the event handlers only for the roving focus one 👍

File: packages/react/toolbar/src/Toolbar.tsx [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+      role="group"
+      data-orientation={orientation}
+      {...props}
+      selector={getSelector(GROUP_NAME)}
+      ref={forwardedRef}
+    />
+  );
+}) as ToolbarGroupPrimitive;
+
+ToolbarGroup.displayName = GROUP_NAME;
```
Comment (benoitgrelard) [inline packages/react/toolbar/src/Toolbar.tsx:110]: I thought we decided to remove this one? Is it a bad merge/rebase?

File: packages/react/toolbar/src/Toolbar.tsx [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+const ToolbarRadioItem = React.forwardRef((props, forwardedRef) => (
+  <RadioGroupPrimitive.Item
+    {...props}
+    as={ToolbarButton}
+    selector={getSelector(RADIO_GROUP_ITEM_NAME)}
+    ref={forwardedRef}
+  />
+)) as ToolbarRadioPrimitive;
+
+ToolbarRadioItem.displayName = RADIO_GROUP_ITEM_NAME;
```
Comment (benoitgrelard) [inline packages/react/toolbar/src/Toolbar.tsx:261]: If we're not doing anything special, should we use the `extendComponent` util?

Comment (jjenzz) [reply packages/react/toolbar/src/Toolbar.tsx:261]: I am doing `as={ToolbarButton}` here and merging types which is why I needed to expand it like this

Comment (benoitgrelard) [reply packages/react/toolbar/src/Toolbar.tsx:261]: Oh I missed that. Does that mean that `as` is taken then and can't be used on `ToolbarRadioItem`?

Comment (jjenzz) [reply packages/react/toolbar/src/Toolbar.tsx:233]: Ah, good point, let me fix that ...

Comment (jjenzz) [reply packages/react/toolbar/src/Toolbar.tsx]: Fixed :+1: good spot 👀

**Review by colmtuite — APPROVED** [review]

**Review by benoitgrelard — APPROVED** [review]

**Pull request merged**